compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2657,6 +2660,20 @@ private boolean checkUnionHasSameFiniteType(LinkedHashSet<BType> memberTypes, BT
         return false;
     }
 
+    private boolean checkValueSpaceHasSameType(BFiniteType finiteType, BType baseType) {
+        if (baseType.tag == TypeTags.FINITE) {
+            return finiteType == baseType;
```

Comment:
Won't work for something like the following, right?

```ballerina
type Foo 1|2;
type Bar 10|11;

public function main() {
    Foo f = 1;
    Bar b = 10;

    boolean x = f < b;
}
```

Reply:
Fixed in 4bbf165